components/brave_wallet/browser/polkadot/polkadot_wallet_service.cc

```diff
@@ -10,19 +10,75 @@
 #include "brave/components/brave_wallet/common/common_utils.h"
 
 namespace brave_wallet {
+namespace {
+
+base::expected<PolkadotChainMetadata, std::string> ParseChainMetadataReponse(
+    const std::optional<std::string>& chain_name,
+    const std::optional<std::string>& err_str) {
```

Comment:
Use `base::optional_ref<std::string>` here, as this function is only called by `OnInitializeChainMetadata`.